system-int-test/test/cmr/system_int_test/search/collection_doi_search_test.clj

```diff
@@ -50,4 +64,29 @@
        [coll1 coll2] ["Doi*"] {:pattern true}
 
        "search for collections with both doi1 and doi2 returns nothing"
-       [] ["Doi1" "doI2"] {:and true}))))
+       [] ["Doi1" "doI2"] {:and true}))
+    (testing "search doi with json query"
+      (are3 [items json-search]
+            (d/refs-match? items (search/find-refs-with-json-query
+                                  :collection {} json-search))
+            "Search for empty values"
+            [coll3] {:doi {:value "Not provided"}}
+
+            "Wildcard search"
+            [coll1 coll2 coll3] {:doi {:value "*" :pattern true}}
+
+            "Search for collections with doi1 or doi2"
```

Comment:
For completeness, add a test case for pattern false.

Reply:
Done